tests/test_constraint_manager.py

```diff
@@ -0,0 +1,348 @@
+        """
+        Test multi-model with both styles of constraints
+        """
+        config = self._create_multi_model_both_constraints()
+        constraints = ConstraintManager.get_constraints_for_all_models(config)
+
+        self.assertEqual(constraints['model_A'],
+                         {'perf_latency_p99': {
+                             'max': 50
+                         }})
+        self.assertEqual(constraints['model_B'],
+                         {'perf_throughput': {
+                             'min': 100
+                         }})
+        self.assertEqual(constraints['model_C'], {
+            'gpu_used_memory': {
+                'max': 50
+            },
+            'perf_throughput': {
+                'min': 50
+            }
+        })
+        self.assertEqual(constraints['model_D'], {
+            'perf_latency_p99': {
+                'max': 100
+            },
+            'gpu_used_memory': {
+                'max': 100
+            }
+        })
+
+    def test_single_model_max_constraint_checks(self):
+        """
+        Test that satisfies_constraints works for a single model
+        with a max style constraint
+        """
+        config = self._create_single_model_with_constraints()
+        constraints = [
+            ConstraintManager.get_constraints_for_all_models(config)['model_A']
+        ]
+
+        # Constraint is P99 Latency max of 100
+        rcm = self._construct_rcm({"perf_latency_p99": 101})
+        self.assertFalse(
+            ConstraintManager.satisfies_constraints(constraints, rcm))
+
+        rcm = self._construct_rcm({"perf_latency_p99": 100})
+        self.assertTrue(
+            ConstraintManager.satisfies_constraints(constraints, rcm))
+
+        rcm = self._construct_rcm({"perf_latency_p99": 99})
+        self.assertTrue(
+            ConstraintManager.satisfies_constraints(constraints, rcm))
+
+    def test_single_model_min_constraint_checks(self):
+        """
+        Test that satisfies_constraints works for a single model
+        with a min style constraint
+        """
+        config = self._create_single_model_global_constraints()
+        constraints = [
+            ConstraintManager.get_constraints_for_all_models(config)['model_A']
+        ]
+
+        # Constraint is throughput min of 100
+        rcm = self._construct_rcm({"perf_throughput": 101})
+        self.assertTrue(
+            ConstraintManager.satisfies_constraints(constraints, rcm))
+
+        rcm = self._construct_rcm({"perf_throughput": 100})
+        self.assertTrue(
+            ConstraintManager.satisfies_constraints(constraints, rcm))
+
+        rcm = self._construct_rcm({"perf_throughput": 99})
+        self.assertFalse(
+            ConstraintManager.satisfies_constraints(constraints, rcm))
+
+    def test_multi_model_constraint_checks(self):
+        """
+        Test that satisfies_constraints works for a multi model
+        with a max style constraint
```

Comment:
is this comment accurate? I see both min and max in the code

Reply:
No. Originally I was going to have separate tests, but decided it was easier to just have one test. Good catch.